src/core/scenario.jl

```diff
@@ -82,6 +82,17 @@ function Scenario(d::Dict; flex_hvac_from_json=false)
     end
     
     site = Site(;dictkeys_tosymbols(d["Site"])...)
+
+    # Check that only PV, storage, and generator are modeled for off-grid
+    if settings.off_grid_flag
+        for key in ["HotThermalStorage", "ColdThermalStorage", "Wind", "FlexibleHVAC", 
```

Comment:
we could make this require less maintenance (i.e. not have to add items to this list) by using a setdiff of keys(d) and the allowed off grid techs:
```julia
julia> offgrid_techs = ["PV", "ElectricStorage", "Generator"]
3-element Vector{String}:
 "PV"
 "ElectricStorage"
 "Generator"

julia> user_keys = ["PV", "CHP"]
2-element Vector{String}:
 "PV"
 "CHP"
 
 julia> unallowed_keys = setdiff(user_keys, offgrid_techs)
1-element Vector{String}:
 "CHP"
 ```
 Then have `if !isempty(unallowed_keys) ...`